src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Base64Url/Base64UrlValidator.cs

```diff
@@ -0,0 +1,91 @@
+        /// <returns><see langword="true"/> if <paramref name="base64UrlText"/> contains a valid, decodable sequence of base-64 encoded data; otherwise, <see langword="false"/>.</returns>
+        /// <remarks>TODO : Update remarks
+        /// If the method returns <see langword="true"/>, the same text passed to <see cref="Convert.FromBase64String(string)"/> and
+        /// <see cref="Convert.TryFromBase64Chars"/> would successfully decode (in the case
+        /// of <see cref="Convert.TryFromBase64Chars"/> assuming sufficient output space). Any amount of whitespace is allowed anywhere in the input,
+        /// where whitespace is defined as the characters ' ', '\t', '\r', or '\n'.
+        /// </remarks>
+        public static bool IsValid(ReadOnlySpan<char> base64UrlText, out int decodedLength) =>
+            Base64.IsValid<char, Base64UrlCharValidatable>(base64UrlText, out decodedLength);
+
+        /// <summary>Validates that the specified span of UTF-8 text is comprised of valid base-64 encoded data.</summary>
+        /// <param name="utf8Base64UrlText">A span of UTF-8 text to validate.</param>
+        /// <returns><see langword="true"/> if <paramref name="utf8Base64UrlText"/> contains a valid, decodable sequence of base-64 encoded data; otherwise, <see langword="false"/>.</returns>
+        /// <remarks> TODO : Update remarks
+        /// where whitespace is defined as the characters ' ', '\t', '\r', or '\n' (as bytes).
+        /// </remarks>
+        public static bool IsValid(ReadOnlySpan<byte> utf8Base64UrlText) =>
+            Base64.IsValid<byte, Base64UrlByteValidatable>(utf8Base64UrlText, out _);
+
+        /// <summary>Validates that the specified span of UTF-8 text is comprised of valid base-64 encoded data.</summary>
+        /// <param name="utf8Base64UrlText">A span of UTF-8 text to validate.</param>
+        /// <param name="decodedLength">If the method returns true, the number of decoded bytes that will result from decoding the input UTF-8 text.</param>
+        /// <returns><see langword="true"/> if <paramref name="utf8Base64UrlText"/> contains a valid, decodable sequence of base-64 encoded data; otherwise, <see langword="false"/>.</returns>
+        /// <remarks> TODO : Update remarks
+        /// where whitespace is defined as the characters ' ', '\t', '\r', or '\n' (as bytes).
+        /// </remarks>
+        public static bool IsValid(ReadOnlySpan<byte> utf8Base64UrlText, out int decodedLength) =>
+            Base64.IsValid<byte, Base64UrlByteValidatable>(utf8Base64UrlText, out decodedLength);
+
+        private static bool ValidateAndDecodeLength(int length, int paddingCount, out int decodedLength)
+        {
+            // Padding is optional for Base64Url, so need to account remainder. If remainder is 1, then it's invalid.
+            int remainder = length % 4;
```

Comment:
```suggestion
            int remainder = (int)((uint)length % 4);
```
For why see this [sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIGYACMhgYQYG8aHunGBLAHYYGAWXIAKQcIA2MAQHMMACwCUDALwA+BrIXKGAUgYoA3Fx70GU0aUlCdcxao3a7GFePEBXKSt1PDYxUzagBfIA==).